### ISSUE TITLE
Out of memory: Killed process xxxx (gcsf) 

### ISSUE THREAD

**Issue by ddonindia** [harababurel/gcsf#77]
While copying large (4GB) file from mounted folder to my local disk I am seeing OOM kill.

Looks like gcsf is keeping all files in memory. 
I tried reducing values of cache_max_seconds , cache_max_items and cache_statfs_seconds but noo luck.

**Comment by harababurel** [comment]
That's true, and unfortunately there's no easy fix since this issue stems from a design decision.

**Comment by ddonindia** [comment]
Ok. 
Sorry mate I had to switched to rclone

**Comment by sanderPostma** [comment]
I'll take a look into the source code when I get to it. On a Raspberry PI this is a real issue since there is not lot of memory available. I have a 750MBit fiber line so mounting a Google drive would make sense for storage that's not speed critical. (I'm paying a lot annually for that 2TB anyway so buying more storage seems a waste of money.)

**Comment by martinszy** [comment]
I think rclone chunks the files every 10mb to avoid this issue. Is it maybe possible to implement chunks in gcsf too?

**Comment by harababurel** [comment]
It might be a possible solution to explore. For now you might be able to get around by setting a trivial cache limit (`cache_max_items = 0`, `cache_max_seconds = 0`). The only limitation with this is that you can only access a file if it would fit in your available RAM.
